From 94428365ea949ec858a3c992548d2379b2027b4b Mon Sep 17 00:00:00 2001
From: Frank Hunleth <fhunleth@troodon-software.com>
Date: Mon, 21 Apr 2014 21:53:40 -0400
Subject: [PATCH] qt5base: enable desktop opengl if available

This adds support for OpenGL in Qt for configurations that include
OpenGL, but not EGL.

Signed-off-by: Frank Hunleth <fhunleth@troodon-software.com>
---
 package/qt5/qt5base/qt5base.mk | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/package/qt5/qt5base/qt5base.mk b/package/qt5/qt5base/qt5base.mk
index 618251e..2e7c2ea 100644
--- a/package/qt5/qt5base/qt5base.mk
+++ b/package/qt5/qt5base/qt5base.mk
@@ -108,7 +108,14 @@ QT5BASE_EGLFS_PLATFORM_HOOKS_SOURCES = \
 	$(@D)/mkspecs/devices/linux-rasp-pi-g++/qeglfshooks_pi.cpp
 endif
 else
-QT5BASE_CONFIGURE_OPTS += -no-opengl -no-eglfs
+QT5BASE_CONFIGURE_OPTS += -no-eglfs
+ifeq ($(BR2_PACKAGE_HAS_LIBGL),y)
+QT5BASE_CONFIGURE_OPTS += -opengl
+QT5BASE_DEPENDENCIES   += libgl
+else
+QT5BASE_CONFIGURE_OPTS += -no-opengl
+endif
+
 endif
 
 QT5BASE_CONFIGURE_OPTS += $(if $(BR2_PACKAGE_OPENSSL),-openssl,-no-openssl)
-- 
1.8.3.2

